File: 1_ReproducibleWorkspace/src/hello.js
// Simple demo module — proves Node.js toolchain is functional.

function greet(name) {
  return `Hello from Node.js ${process.version}, ${name}!`;
}

function envInfo() {
  return {
    node: process.version,
    platform: process.platform,
    arch: process.arch,
  };
}

if (require.main === module) {
  console.log(greet("poserForge"));
  const info = envInfo();
  for (const [key, val] of Object.entries(info)) {
    console.log(`  ${key}: ${val}`);
  }
}

module.exports = { greet, envInfo };

File: 1_ReproducibleWorkspace/src/hello.py
"""Simple demo module — proves Python toolchain is functional."""

import sys
import platform


def greet(name: str) -> str:
    return f"Hello from Python {sys.version_info.major}.{sys.version_info.minor}, {name}!"


def env_info() -> dict:
    return {
        "python": sys.version,
        "platform": platform.system(),
        "arch": platform.machine(),
    }


if __name__ == "__main__":
    print(greet("poserForge"))
    for key, val in env_info().items():
        print(f"  {key}: {val}")
